### ISSUE TITLE
Research adding configuration option for setting the position of the overlay timer

### ISSUE THREAD

**Issue by jmakhack** [jmakhack/hydrate-reminder#142]
## Feature Reasoning

The overlay timer (when enabled) is displayed in the upper left corner of the screen. Not all users my want it displayed in that location so having a configuration option to be able to display it on other parts of the UI would be nice.

## Feature Description

This enhancement is to add a configuration option that presents a drop-down list of various locations where the overlay timer can be displayed. By default, the timer should be displayed in the upper left where it is currently being displayed.

Screenshot of current position for reference:
![position](https://user-images.githubusercontent.com/1442227/177127481-98109c42-3822-49cf-a4e6-d4cdf98de8dd.png)

## Possible Alternatives

Part of the work for this feature is to determine what options are even available for the position of the overlay timer. If there are no other possible options, then this feature would have to be scrapped. If there are a very high number of options, it would be nice to limit the number of options as to not overwhelm the user with too many choices to select from.

If other configuration possibilities related to the overlay timer are discovered while investigating what is available, it would be good to track them in the comments below for consideration on future feature requests.

## Additional Context

It is important to add unit tests where possible when doing the implementation work for this feature.

**Comment by jmakhack** [comment]
Changing this to a research task and will open a separate issue for the implementation work with a more detailed explanation on how to implement the feature.

**Comment by jmakhack** [comment]
Doing a quick search through the RuneLite APIs, I was able to find the classes and methods for handling overlay logic here: https://static.runelite.net/api/runelite-client/net/runelite/client/ui/overlay/package-summary.html

In particular the `OverlayPosition` enum would be most important in the investigation on how to implement this feature: https://static.runelite.net/api/runelite-client/net/runelite/client/ui/overlay/OverlayPosition.html

The enum value would then be used by the `setPosition` method of the `Overlay` class: https://static.runelite.net/api/runelite-client/net/runelite/client/ui/overlay/Overlay.html

The `Overlay` class also contains other position altering methods such as `setSnappable`, `setPreferredPosition`, `setDragTargetable`, and `setMovable` that should be explored as well. It would be good to also investigate potential options around the `OverlayLayer` and `OverlayPriority` as well and see how those options can be used to improve the overall usability and customizability of this plugin.

**Comment by jmakhack** [comment]
Closing this out for now as I've decided this is out of scope for v2.0.0. The overlay timer would need some considerable amount of refactoring to support this functionality. This is because the overlay timer extends from the `Infobox` class which does not support repositioning outside of being in the upper-left corner.

However, if time permits and everything else is finished considerably early, I may reopen this for v2.0.0. It would just be considered low priority and a extra nice to have feature.
